src/lsm6.ts

```diff
@@ -0,0 +1,305 @@
+    OUTX_H_G          = 0x23,
+    OUTY_L_G          = 0x24,
+    OUTY_H_G          = 0x25,
+    OUTZ_L_G          = 0x26,
+    OUTZ_H_G          = 0x27,
+    OUTX_L_XL         = 0x28,
+    OUTX_H_XL         = 0x29,
+    OUTY_L_XL         = 0x2A,
+    OUTY_H_XL         = 0x2B,
+    OUTZ_L_XL         = 0x2C,
+    OUTZ_H_XL         = 0x2D,
+
+    FIFO_STATUS1      = 0x3A,
+    FIFO_STATUS2      = 0x3B,
+    FIFO_STATUS3      = 0x3C,
+    FIFO_STATUS4      = 0x3D,
+    FIFO_DATA_OUT_L   = 0x3E,
+    FIFO_DATA_OUT_H   = 0x3F,
+    TIMESTAMP0_REG    = 0x40,
+    TIMESTAMP1_REG    = 0x41,
+    TIMESTAMP2_REG    = 0x42,
+
+    STEP_TIMESTAMP_L  = 0x49,
+    STEP_TIMESTAMP_H  = 0x4A,
+    STEP_COUNTER_L    = 0x4B,
+    STEP_COUNTER_H    = 0x4C,
+
+    FUNC_SRC          = 0x53,
+
+    TAP_CFG           = 0x58,
+    TAP_THS_6D        = 0x59,
+    INT_DUR2          = 0x5A,
+    WAKE_UP_THS       = 0x5B,
+    WAKE_UP_DUR       = 0x5C,
+    FREE_FALL         = 0x5D,
+    MD1_CFG           = 0x5E,
+    MD2_CFG           = 0x5F,
+}
+
+const DS33_WHO_ID = 0x69;
+
+// Range scales for the accelerometer
+export enum AccelerometerScale {
+    SCALE_2G = "SCALE_2G",
+    SCALE_4G = "SCALE_4G",
+    SCALE_8G = "SCALE_8G",
+    SCALE_16G = "SCALE_16G"
+}
+
+// Range scales for the gyro
+export enum GyroScale {
+    SCALE_125_DPS = "SCALE_125_DPS",
+    SCALE_250_DPS = "SCALE_250_DPS",
+    SCALE_500_DPS = "SCALE_500_DPS",
+    SCALE_1000_DPS = "SCALE_1000_DPS",
+    SCALE_2000_DPS = "SCALE_2000_DPS"
+}
+
+export interface Vector3 {
+    x: number;
+    y: number;
+    z: number;
+}
+
+const ACCEL_SCALE_CTRL_BYTE: Map<AccelerometerScale, number> = new Map<AccelerometerScale, number>();
+ACCEL_SCALE_CTRL_BYTE.set(AccelerometerScale.SCALE_2G, 0x80);
+ACCEL_SCALE_CTRL_BYTE.set(AccelerometerScale.SCALE_4G, 0x88);
+ACCEL_SCALE_CTRL_BYTE.set(AccelerometerScale.SCALE_8G, 0x8C);
+ACCEL_SCALE_CTRL_BYTE.set(AccelerometerScale.SCALE_16G, 0x84);
+
+const GYRO_SCALE_CTRL_BYTE: Map<GyroScale, number> = new Map<GyroScale, number>();
+GYRO_SCALE_CTRL_BYTE.set(GyroScale.SCALE_125_DPS, 0x82);
+GYRO_SCALE_CTRL_BYTE.set(GyroScale.SCALE_250_DPS, 0x80);
+GYRO_SCALE_CTRL_BYTE.set(GyroScale.SCALE_500_DPS, 0x84);
+GYRO_SCALE_CTRL_BYTE.set(GyroScale.SCALE_1000_DPS, 0x88);
+GYRO_SCALE_CTRL_BYTE.set(GyroScale.SCALE_2000_DPS, 0x8C);
+
+// Sensitivity in mg/LSB
+const ACCEL_OUTPUT_SCALE_FACTOR: Map<AccelerometerScale, number> = new Map<AccelerometerScale, number>();
+ACCEL_OUTPUT_SCALE_FACTOR.set(AccelerometerScale.SCALE_2G, 0.061);
+ACCEL_OUTPUT_SCALE_FACTOR.set(AccelerometerScale.SCALE_4G, 0.122);
+ACCEL_OUTPUT_SCALE_FACTOR.set(AccelerometerScale.SCALE_8G, 0.244);
+ACCEL_OUTPUT_SCALE_FACTOR.set(AccelerometerScale.SCALE_16G, 0.488);
+
+// Sensitivity in mdps/LSB
+const GYRO_OUTPUT_SCALE_FACTOR: Map<GyroScale, number> = new Map<GyroScale, number>();
+GYRO_OUTPUT_SCALE_FACTOR.set(GyroScale.SCALE_125_DPS, 4.375);
+GYRO_OUTPUT_SCALE_FACTOR.set(GyroScale.SCALE_250_DPS, 8.75);
```

Comment:
Same here. I would expect this to be 245 not 250.

Reply:
It's referred to as 250dps in the datasheet